src/data_provider/src/extended_sources/browser_extensions/src/chrome.cpp

```diff
@@ -0,0 +1,685 @@
+/* Copyright (C) 2015, Wazuh Inc.
+ * All rights reserved.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+
+#include "chrome.hpp"
+#include <tuple>
+#include <iostream>
+#include <fstream>
+#include <algorithm>
+#include "openssl/sha.h"
+#include "stringHelper.h"
+#include "filesystemHelper.h"
+
+#define MAX_PATH_LENGTH 1000
+
+namespace chrome
+{
+    ChromeExtensionsProvider::ChromeExtensionsProvider(std::shared_ptr<IChromeExtensionsWrapper> chromeExtensionsWrapper) : m_chromeExtensionsWrapper(std::move(chromeExtensionsWrapper))
+    {
+    }
+
+    ChromeExtensionsProvider::ChromeExtensionsProvider() : m_chromeExtensionsWrapper(std::make_shared<ChromeExtensionsWrapper>())
+    {
+    }
+
+    bool ChromeExtensionsProvider::isValidChromeProfile(const std::string& profilePath)
```

Comment:
Input Validation for Path Safety

```suggestion
    bool ChromeExtensionsProvider::isValidChromeProfile(const std::string& profilePath)
  {
     if (profilePath.empty() ||
          profilePath.find("..") != std::string::npos ||
          profilePath.length() > 4096)
      {
          return false;
      }

      return Utils::existsRegular(Utils::joinPaths(profilePath, kPreferencesFile)) ||
             Utils::existsRegular(Utils::joinPaths(profilePath, kSecurePreferencesFile));
  }

```